name: Prusa error codes

on:
  push:
  pull_request:

jobs:
  tests:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        python-version: [ '3.10', '3.11', '3.12' ]

    steps:
    - uses: actions/checkout@v4
    - name: Set up Python
      uses: actions/setup-python@v5
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install pylint
        pip install coverage
        pip install pyyaml
        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
    - name: Lint with pylint
      id: pylint
      run: |
        python3 -m pylint --version
        python3 -m pylint --persistent=n --jobs 0 --max-line-length=120 prusaerrors
    - name: Test
      # Run the step even if the pylint step has failed
      if: success() || (failure() && steps.pylint.conclusion == 'failure')
      run: |
        export PATH="${PATH}:$(pwd)"
        coverage run -m unittest discover --failfast --verbose tests
        coverage report

